### DISCUSSION TITLE
How to redirect to resource only route that returns HTML?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#3405] [Q&A] [answered]
This is a pretty odd question, but I have the following route:

```typescript
export const loader: LoaderFunction = async ({ params }) => {
  const postId = getParam(params, "postId") as PostId;
  return new Response(`<!doctype html> <html> <body>Post id is: ${postId}</body></html>`, {
    headers: {
      "Content-Type": "text/html",
    },
  });
};
```

I know this defeats the entire point of Remix, since I'm literally templating HTML, but there's a good reason. The full example is more complex. (I'm basically embedding another site after modifying its HTML on the server, it's almost like I turned a Chrome extension into a web page).

This works fine, but I have an issue. If I do: `return redirect("/p" + postId);` in the action/loader of a different route, Remix will crash.
This makes sense because Remix is hydrating my application into an SPA, then (presumably) it intercepts the redirect and tries to do its own SPA logic, which fails because I am going to a un-hydrated, non-remixified page.

What is a good workaround here? (Asides from sticking all the HTML in an iFrame, which I could do, but I don't want to do).

**Comment by sergiodxa** [comment, marked as answer]
You could add `reloadDocument` to the Link or Form who triggered the loader/action that does the redirect, that way Remix will not navigate like a SPA
